man/systemd.service.xml

```diff
@@ -153,77 +153,89 @@
+            used in combination with <varname>PrivateNetwork=</varname><option>yes</option>.</para></listitem>
+
+            <listitem><para>Behavior of <option>idle</option> is very similar to <option>simple</option>; however,
+            actual execution of the service program is delayed until all active jobs are dispatched. This may be used
+            to avoid interleaving of output of shell services with the status output on the console. Note that this
+            type is useful only to improve console output, it is not useful as a general unit ordering tool, and the
+            effect of this service type is subject to a 5s time-out, after which the service program is invoked
+            anyway.</para></listitem>
+          </itemizedlist>
+
+          <para>It is generally recommended to use <varname>Type=</varname><option>simple</option> for long-running
+          services whenever possible, as it is the simplest and fastest option. However, as this service type won't
+          propagate service start-up failures and doesn't allow ordering of other units against completion of
+          initialization of the service (which for example is useful if clients need to connect to the service through
+          some form of IPC, and the IPC channel is only established by the service itself — in contrast to doing this
+          ahead of time through socket or bus activation or similar), it might not be sufficient for many cases. If so,
+          <option>notify</option> or <option>dbus</option> (the latter only in case the service provides a D-Bus
+          interface) is the preferred option as it allows service program code to precisely schedule when to consider
+          the service started up successfully and when to proceed with follow-up units. The <option>notify</option>
+          service type requires explicit support in the service codebase (as <function>sd_notify()</function> or an
+          equivalent API needs to be invoked by the service at the appropriate time) — if it's not supported, then
+          <option>forking</option> is an alternative: it supports the traditional UNIX service start-up
+          protocol. Finally, <option>exec</option> might be an option for cases where it is enough to ensure the
+          service binary is invoked, and where the service binary itself executes no or little initialization on its
+          own (and its initialization is unlikely to fail). Note that using any type other than <option>simple</option>
+          possibly delays the boot process, as the service manager needs to wait for service initialization to
+          complete. It is hence recommended not to needlessly use any types other than <option>simple</option>. (Also
+          note it is generally not recommended to use <option>idle</option> or <option>oneshot</option> for
+          long-running services.)</para>
```

Comment:
Hmmm, I think we should instead recommend "Type=exec" for normal service, just because that gives much better reporting when starting from the command line, which is more important for users than a µs here and there. But let's see how exec behaves, and we can always change that in the future.